src/invidious/routes/embed.cr

```diff
@@ -1,5 +1,5 @@
-class Invidious::Routes::Embed < Invidious::Routes::BaseRoute
-  def redirect(env)
+module Invidious::Routes::Embed
+  def self.redirect(env)
```

Comment:
Why is `self.` required in modules? I'm not sure to understand that correctly.

Reply:
Modules cannot be initialized so aren't able to use the defined instance methods of that module. Hence why we have to use class methods.

In PR #2277 you used an `extend self` keyword on that module.

```crystal
module YoutubeAPI
  extend self
```

And [an extend makes a type include methods defined in that module as class methods](https://crystal-lang.org/reference/syntax_and_semantics/modules.html).

Reply:
Oooh, yeah, right. So adding `self.` on all methods or `extend self` at the top is basically the same, right?

Reply:
Yep!